File: data-structures-and-algorithms-ashrafi/python/src/recursive/problem_2/problem_2.py
def average(a, l, i):
    if i == l - 1:
        return a[i] / l
    return (a[i] / l) + average(a, l, i + 1) 


def main():
    nums = [2, 1, 4, 3, 6, 5]
    length = len(nums)
    i = 0
    
    print(f"Average = {average(nums, length, i)}")


if __name__ == "__main__":
    main()

File: data-structures-and-algorithms-ashrafi/java/src/recursive/problem2/Problem2.java
public class Problem2 {

    public static double average(int[] a, int l, int i){
        if (i == l - 1) {
            return (double)a[i] / l;
        }    
        return ((double)a[i] / l) + average(a, l, i + 1);
    }

    public static void main(String[] args) {
        int[] nums = {2, 1, 4, 3, 6, 5};
        int lenght = nums.length;
        int i = 0;

        System.out.printf("Average = %.2f", average(nums, lenght, i));
    }
}